DBoW2/DBoW2_sample_code.cpp

```diff
@@ -0,0 +1,225 @@
+#include "DBoW2.h"
+#include <opencv2/core.hpp>
+#include <opencv2/highgui.hpp>
+#include <opencv2/features2d.hpp>
+#include <easy/profiler.h>
+
+#include <iostream>
+#include <vector>
+#include <filesystem>
+#include <string>
+#include <memory>
+
+using namespace DBoW2;
+namespace fs = std::filesystem;
+namespace SampleCode
+{
+  using FeatureVector = std::vector<std::vector<cv::Mat>>;
+  using Features = std::vector<cv::Mat>;
+  using FileLUT = std::vector<std::string>;
+}
+
+bool loadDBFeatures(SampleCode::FeatureVector& vfeatures, SampleCode::FileLUT& dbTable, const cv::Ptr<cv::ORB>& orb);
+bool loadVocabulary(OrbVocabulary& voc);
+void changeStructure(const cv::Mat& plain, SampleCode::Features& out);
+bool createDatabase(OrbDatabase& db, const SampleCode::FeatureVector& vfeatures);
+bool loadQueryFeatures(SampleCode::FeatureVector& vfeatures, const cv::Ptr<cv::ORB>& orb);
+bool queryDatabase(OrbDatabase& db, const SampleCode::FeatureVector& vfeatures, SampleCode::FileLUT& queryResultTable, const SampleCode::FileLUT& dbTable);
+bool saveResult(const SampleCode::FileLUT& queryResultTable);
+
+
+// Number of Images to Build Database
+constexpr int NDBIMAGES = 4;
+// Number of Images to Query with Database
+constexpr int NQUERYIMAGES = 1;
+
+
+// Paths 
+fs::path curPath = fs::current_path();
+fs::path vocPath = curPath.parent_path() / "ORBvoc" / "ORBvoc.txt";
+fs::path dbPath = curPath.parent_path() / "demo" / "Database";
+fs::path queryPath = curPath.parent_path() / "demo" / "Query";
+fs::path savePath = curPath.parent_path() / "demo" / "Result" / "";
+
+
+
+int main()
+{
+  EASY_MAIN_THREAD;
+  EASY_PROFILER_ENABLE;
+
+  // DataStructures
+  SampleCode::FileLUT queryResultTable;
+  SampleCode::FileLUT dbTable;
+  SampleCode::FeatureVector dbFeatures;
+  SampleCode::FeatureVector queryFeatures;
+  cv::Ptr<cv::ORB> ORB = cv::ORB::create();
+  auto ptrVocabulary = std::make_unique<OrbVocabulary>();
+
+  if(!loadDBFeatures(dbFeatures, dbTable, ORB))
+    std::cerr << "Error while loading DB features\n" << std::endl;
+
+  if(!loadVocabulary(*ptrVocabulary))
+    std::cerr << "Error while loading Vocabulary\n" << std::endl;
+
+  EASY_BLOCK("Initialize Database", profiler::colors::LightBlue);
+  auto ptrDatabase = std::make_unique<OrbDatabase>(*ptrVocabulary);

```

Comment:
```suggestion
  auto& vocabulary = *ptrVocabulary;
  auto ptrDatabase = std::make_unique<OrbDatabase>(vocabulary);

```
가독성을 높이기 위해 위와 같은 변화를 제안합니다.
어떻게 생각하시나요?

Reply:
음 처음에는 de-reference를 사용하여 한 줄로 작성하는게 보기에 깔끔하다고 생각했었는데, 
형기님이 제안하신 코드를 보니 코드를 보는 사람 입장에서 '이해'하기에는 두 줄로 구분해서 Vocabulary를 사용해 Database를 구축한다는 의미가 더 쉽게  와닿을 수 있겠다는 생각이 드네요!! 
반영해서 수정하도록 하겠습니다~ 감사합니다!